### PR TITLE
TG-1225 Update `webview_flutter_wkwebview` to version with Privacy Manifests

### PR DESCRIPTION
**Pull request by mark-hkz**
Task [TG-1225 Update `webview_flutter_wkwebview` to version with Privacy Manifests](https://taiga.hackazouk.work/project/official-alumni-mobile/task/1225?)
---
- update webview_flutter plugin to version 4.6.0 to support apple's [Upcoming third-party SDK requirements](https://developer.apple.com/support/third-party-SDK-requirements/)

### CODE REVIEW

File: lib/src/widgets/iframe_mobile.dart [hunk 1]
```diff
@@ -27,35 +28,100 @@ class IframeContentElement extends ReplacedElement {
...
+  final UniqueKey key = UniqueKey();
+
+  @override
+  void initState() {
+    super.initState();
+    if (webview.WebViewPlatform.instance is WebKitWebViewPlatform) {
+      params = WebKitWebViewControllerCreationParams(
+        allowsInlineMediaPlayback: true,
+        mediaTypesRequiringUserAction: const <PlaybackMediaTypes>{},
+      );
```
Comment (m-sugawara) [inline lib/src/widgets/iframe_mobile.dart]: This setting does not seem to exist before the migration. Could you let me know why it is needed? 🙇

File: lib/src/widgets/iframe_mobile.dart [hunk 2]
```diff
@@ -27,35 +28,100 @@ class IframeContentElement extends ReplacedElement {
...
+  final NavigationDelegate? navigationDelegate;
+
+  @override
+  State<_WebViewWidget> createState() => _WebViewWidgetState();
+}
+
+class _WebViewWidgetState extends State<_WebViewWidget> {
+  late final webview.WebViewController controller;
+  late final webview.PlatformWebViewControllerCreationParams params;
+  final UniqueKey key = UniqueKey();
```
Comment (m-sugawara) [inline lib/src/widgets/iframe_mobile.dart:65]: [nit] With this change, `IframeContentElement.key` seems to no longer be used.

**Review by m-sugawara — COMMENTED** [review]
@mark-hkz 
Is this the correct document you referred to during the migration? Please let me know if there are others.

https://github.com/flutter/packages/tree/webview_flutter-v4.6.0/packages/webview_flutter/webview_flutter#migrating-from-30-to-40

File: lib/src/widgets/iframe_mobile.dart (hunk 1 above)
Comment (mark-hkz) [reply lib/src/widgets/iframe_mobile.dart]: @m-sugawara Thanks for the review 🙏 
- this is not needed 🙇 
- i have updated the PR to remove the unnecessary changes.

**Comment by mark-hkz** [comment]
> @mark-hkz Is this the correct document you referred to during the migration? Please let me know if there are others.
> 
> https://github.com/flutter/packages/tree/webview_flutter-v4.6.0/packages/webview_flutter/webview_flutter#migrating-from-30-to-40

Yes this is the document Im referring to, `webview_flutter v4.6.0` uses `webview_flutter_wkwebview v3.11.0` which has the privacy manifest changes ([ref](https://pub.dev/packages/webview_flutter_wkwebview/changelog))

**Review by m-sugawara — APPROVED** [review]
Thank you for your response!
LGTM 👍

**Review by gino2527 — APPROVED** [review]

**Pull request merged**